Physical_Engine/Query/BottomCentreline.cs

```diff
@@ -0,0 +1,96 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.Engine.Geometry;
+using BH.Engine.Reflection;
+using BH.Engine.Spatial;
+using BH.oM.Base;
+using BH.oM.Geometry;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Spatial.ShapeProfiles;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Configuration;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+
+        [Description("Returns the bottom centreline of an IFramingElement.")]
+        [Input("element", "The IFramingElement to query the bottom centreline of.")]
+        [Output("curve", "The bottom centreline of the IFramingElement.")]
+
+        public static ICurve BottomCentreline(this BH.oM.Physical.Elements.IFramingElement element)
+        {
+            ICurve location = element.Location;
+
+            Vector normal;
+
+            try
+            {
+                normal = BH.Engine.Physical.Query.Normal(element);
+            }
+            catch
+            {
+                Engine.Reflection.Compute.RecordError("IFramingElement must have linear location line.");
+                return null;
+            }
+
+            if (normal == null)
+            {
+                Engine.Reflection.Compute.RecordError("Was not able to compute element normal.");
+                return null;
+            }
+
+            if(element.Property is ConstantFramingProperty)
+            {
+                ConstantFramingProperty constantProperty = element.Property as ConstantFramingProperty;
+
+                IProfile profile = constantProperty.Profile;
+
+                BoundingBox profileBounds = profile.Edges.Bounds();
+
+                Point profileMax = profileBounds.Max;
+
+                Point profileMin = profileBounds.Min;
+
+                double height = profileMax.Y - profileMin.Y;
+
+                ICurve bottomCentreline = location.ITranslate(normal * -0.5 * height);
+
+                return bottomCentreline;
+            }
+            else
+            {
+                Engine.Reflection.Compute.RecordError("Element does not have ConstantFramingProperty, was not able to compute section height.");
```

Comment:
In the future, further framing properties might be added. To avoid this error needing updating every time, I'd advise we go for `Element does not have a suitable framing property, so the section height could not be calculated`.